### PR TITLE
added nil check for kvPair

### PR DESCRIPTION
**Pull request by Michael-Norris**
This change is [Reviewable](https://reviewable.io/reviews/dollarshaveclub/talcum/5)

### CODE REVIEW

File: src/cmd/talcum/main.go [hunk 1]
```diff
@@ -90,13 +90,19 @@ func main() {
...
 	} else if selectorConfigConsulPath != "" {
-		kvpair, _, err := kvClient.Get(selectorConfigConsulPath, nil)
+		kvPair, _, err := kvClient.Get(selectorConfigConsulPath, nil)
 		if err != nil {
 			clierr("error reading consul KV: %v", err)
 		}
-		if err := json.Unmarshal(kvpair.Value, &selectorConfig); err != nil {
-			clierr("error unmarshaling config: %v", err)
+
+		if kvPair != nil {
```
Comment (hankjacobs) [inline src/cmd/talcum/main.go]: inverting this conditional will help avoid the pyramid of if's

**Review by hankjacobs — CHANGES_REQUESTED** [review]

File: src/cmd/talcum/main.go (hunk 1 above)
Comment (bkeroackdsc) [inline src/cmd/talcum/main.go]: I would do something like:

```go
if err != nil || kvPair == nil {
			clierr("error reading consul KV or result is nil: %v", err)
		}
```

**Review by bkeroackdsc — CHANGES_REQUESTED** [review]

File: src/cmd/talcum/main.go [hunk 2]
```diff
@@ -90,11 +90,11 @@ func main() {
 			clierr("error unmarshaling config: %v", err)
 		}
 	} else if selectorConfigConsulPath != "" {
-		kvpair, _, err := kvClient.Get(selectorConfigConsulPath, nil)
-		if err != nil {
+		kvPair, _, err := kvClient.Get(selectorConfigConsulPath, nil)
+		if err != nil || kvPair == nil {
 			clierr("error reading consul KV: %v", err)
```
Comment (hankjacobs) [inline src/cmd/talcum/main.go]: Update error message to include possibility that kvPair was nil

**Review by hankjacobs — CHANGES_REQUESTED** [review]

**Review by Michael-Norris — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dollarshaveclub/talcum/5)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @hankjacobs, @michaelNorrisDollarShaveClub, and @ahamilton55)

___
*[src/cmd/talcum/main.go, line 95 at r2](https://reviewable.io/reviews/dollarshaveclub/talcum/5#-LLpn2Hx1olcxwOJYZJy-r2-95:-LLptqer4ghPHNpTEFW3:b-er90c3) ([raw file](https://github.com/dollarshaveclub/talcum/blob/98057a2d977d4e4eb8cb9a8305babab7aa4cf2b3/src/cmd/talcum/main.go#L95)):*
<details><summary><i>Previously, hankjacobs (Hank Jacobs) wrote…</i></summary><blockquote>

Update error message to include possibility that kvPair was nil
</blockquote></details>

code updated

**Review by hankjacobs — APPROVED** [review]

**Review by Michael-Norris — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dollarshaveclub/talcum/5)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @hankjacobs and @ahamilton55)

___
*[src/cmd/talcum/main.go, line 95 at r2](https://reviewable.io/reviews/dollarshaveclub/talcum/5#-LLpn2Hx1olcxwOJYZJy-r2-95:-LLpv2hN74bkyZD8X9E6:b-896fix) ([raw file](https://github.com/dollarshaveclub/talcum/blob/98057a2d977d4e4eb8cb9a8305babab7aa4cf2b3/src/cmd/talcum/main.go#L95)):*
<details><summary><i>Previously, michaelNorrisDollarShaveClub (Michael Norris) wrote…</i></summary><blockquote>

code updated
</blockquote></details>

Done.

**Review by Michael-Norris — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dollarshaveclub/talcum/5)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @hankjacobs and @ahamilton55)

___
*[src/cmd/talcum/main.go, line 98 at r1](https://reviewable.io/reviews/dollarshaveclub/talcum/5#-LLpn2Hx1olcxwOJYZJy-r1-98:-LLpv3yz3z7AO62yPW6d:b-896fix) ([raw file](https://github.com/dollarshaveclub/talcum/blob/86f51d68a567f26ece4bcf6c84e8dee6e6ba0ffa/src/cmd/talcum/main.go#L98)):*
<details><summary><i>Previously, bkeroackdsc (Benjamen Keroack) wrote…</i></summary><blockquote>

I would do something like:

```go
if err != nil || kvPair == nil {
	clierr("error reading consul KV or result is nil: %v", err)
}
```
</blockquote></details>

Done.

**Pull request merged**